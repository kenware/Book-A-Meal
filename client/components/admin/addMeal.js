import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Dropzone from 'react-dropzone';
import FormData from 'form-data';
import PropTypes from 'prop-types';
import * as mealActions from '../../redux/Action/mealAction';
import * as actions from '../../redux/Action/action';
import './index.scss';

export class Add extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      price: '',
      description: '',
      file: [],
      addMeal: 'Add Meal',
      validDescription: '',
      validPrice: '',
      validName: '',
      image: ''
    };
    this.onChange = this.onChange.bind(this);
    this.addMeal = this.addMeal.bind(this);
    this.onDrop = this.onDrop.bind(this);
  }
  /**
   * lifecycle hook called when component receives props
   *
   * return new state after it is changed
   */
  static getDerivedStateFromProps(props) {
    if (props.errorMessage.createMealError || props.successMessage.createMealSuccess) {
      return { addMeal: 'Add Meal' };
    }
    return null;
  }
  /**
   * remove error props in redux store if component unmount
   */
  componentWillUnmount() {
    this.props.actions.clearMessages();
  }
  /**
   * @param  {} e set state on input onchange event
  */
  onChange(e) {
    const { state } = this;
    state[e.target.name] = e.target.value;
    this.setState(state);
  }
  /**
   * @param  {} file sets image to state using drop box
  */
  onDrop(file) {
    this.setState({ file, image: '' });
  }
  /**
   * calls a redux function
   * adds a meal using form-data
  */
  addMeal(e) {
    e.preventDefault();
    this.props.actions.clearMessages();
    const {
      name, price, description, file
    } = this.state;
    if (!name) { this.setState({ validName: 'Name is required' }); return; }
    if (!price) { this.setState({ validPrice: 'Price is required' }); return; }
    if (!description) { this.setState({ validDescription: 'Description is required' }); return; }
    const payload = new FormData();
    payload.append('name', name);
    payload.append('price', price);
    payload.append('description', description);
    file.forEach((photo) => {
      payload.append('file', photo);
    });
    this.setState({
      addMeal: (<div><i className="fa fa-spinner fa-spin fa-2x fa-fw" aria-hidden="true" /></div>)
    });
    this.props.mealActions.createMeal(payload);
  }
  render() {
    return (
      <div>
        <h2 style={{ marginTop: '2rem' }}>ADD A MEAL </h2>
        <div className="register-container" style={{ marginTop: '2px' }}>
          <div className="register-col" />
          <div className="register-wrapper" style={{ marginTop: '2px' }}>
            <h4 className="p-color text-center">Fill The Field Below To Add A Meal</h4><br />
            <h3 className="text-center danger">{this.props.errorMessage.createMealError }</h3>
            <h3 className="text-center danger">{this.state.message }</h3>
            <form>
              <div className="form-field">
                <label htmlFor="name">Name     <br />
                  <font color="red">{this.state.validName} </font>
                </label>
                <input onChange={this.onChange} type="text" id="name" name="name" placeholder="Jollof Rice" required />
              </div>
              <div className="form-field">
                <label htmlFor="price">Price<br />
                  <font color="red">{this.state.validPrice} </font>
                </label>
                <input onChange={this.onChange} type="number" id="price" name="price" placeholder="5000" required />
              </div>
              <div className="form-field">
                <label htmlFor="name">Description<br />
                  <font color="red">{this.state.validDescription} </font>
                </label>
                <textarea cols="7" rows="9" id="description" onChange={this.onChange} className="form-label descriptions" name="description" placeholder="Description of the meal" required />

              </div>
              <div className="form-field">
                <label htmlFor="password">
                  Image
                </label>
                <span className="form-label">
                  <Dropzone
                    onDrop={this.onDrop}
                    accept="image/jpeg,image/jpg,image/tiff,image/gif,image/png,image/svg"
                    ref="dropzone"
                    multiple={false}
                  >
                    Drag and drop or click to select an image to upload.
                  </Dropzone>
                  {this.state.file.map(fil => <img key={fil.preview} src={fil.preview} className="img-fluid" style={{ width: '200px', height: '200px' }} alt="upload" />)}
                </span>
              </div>
              <div className="form-field">
                <button type="submit" className="button lg submit" name="submit" onClick={this.addMeal}>{this.state.addMeal}</button>
              </div>
              <div className="form-field">
                <span className="text-center form-label p-color">{this.props.successMessage.createMealSuccess }</span>
              </div>
            </form>

          </div>
          <div className="register-col" />
        </div>
      </div>
    );
  }
}
Add.propTypes = {
  errorMessage: PropTypes.object.isRequired,
  successMessage: PropTypes.object.isRequired,
  mealActions: PropTypes.object.isRequired,
  actions: PropTypes.object.isRequired
};
export function mapStateToProps(state) {
  return {
    errorMessage: state.errorMessage,
    successMessage: state.successMessage
  };
}
export function mapDispatchToProps(dispatch) {
  return {
    mealActions: bindActionCreators(mealActions, dispatch),
    actions: bindActionCreators(actions, dispatch)
  };
}
export default connect(mapStateToProps, mapDispatchToProps)(Add);
