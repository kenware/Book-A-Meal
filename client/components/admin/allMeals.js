import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import Modal from 'react-responsive-modal';
import Pagination from 'react-js-pagination';
import './index.scss';
import * as mealActions from '../../redux/Action/mealAction';


const limit = 6;
export class AllMeal extends Component {
  constructor(props) {
    super(props);
    this.state = {
      mealId: 0,
      mealName: '',
      mealImage: '',
      open: false,
      activePage: 1
    };
    this.cancelDelete = this.cancelDelete.bind(this);
    this.deleteMeal = this.deleteMeal.bind(this);
    this.handlePageChange = this.handlePageChange.bind(this);
    this.onOpenModal = this.onOpenModal.bind(this);
  }

  componentDidMount() {
    this.props.mealActions.getAllMeals(limit, 0);
  }

  /**
   * @param  {} mealId
   * @param  {} mealName
   * @param  {} mealImage
   * displays modal that give options to delete a meal or cancel delete
   */
  onOpenModal(mealId, mealName, mealImage) {
    this.setState({
      mealId, mealName, mealImage, open: true
    });
  }
  /**
   * closes modal without deleting a meal
   */
  cancelDelete() {
    this.setState({
      mealId: '', mealName: '', mealImage: '', open: false
    });
  }

  /**
   * calls a redux action
   * deletes a meal
   */
  deleteMeal() {
    if (this.state.mealId) {
      this.props.mealActions.deleteMeal(this.state.mealId);
      this.setState({
        mealId: '', mealName: '', mealImage: '', open: false
      });
    }
  }

  handlePageChange(pageNumber) {
    const offset = (pageNumber - 1) * limit;
    this.setState({ activePage: pageNumber });
    this.props.mealActions.getAllMeals(limit, offset);
  }

  render() {
    return (
      <div id="allmeal">
        <Modal open={this.state.open} onClose={this.cancelDelete} center>
          <br />
          <div className="modal-header">
            <p className="justify l-r-pad-text"> You are about to delete {this.state.mealName}</p>
          </div>
          <div className="modal-contents" style={{ justifyContent: 'center' }}>
            <img src={this.state.mealImage} className="rounded-circle img-height" alt="meal" />
          </div>
          <div className="modal-footer">
            <button className="remove-modal delete-meal" onClick={this.deleteMeal}>Delete</button>
          </div>
        </Modal>
        <h2>All Meal Options{this.props.successMessage.createMealSuccess }</h2>
        <h4 className="p-color text-center">Edit, delete or View all meal options</h4>
        <h2 className="danger">{this.props.successMessage.deleteMessage }</h2>
        {this.props.meals.rows.map(meal =>
        (
          <div className="contents container allMeal" key={meal.id}>
            <div className="content-wrap">
              <div className="col-meal l-r-pad-text">
                <img src={meal.image} className="rounded-circle img-height" alt="meals" />
              </div>
              <div className="col-meal">
                <h4 className="p-color"> Title</h4>
                <Link to={`/admin/edit/${meal.id}`} id="edit">{meal.name}</Link>
              </div>
              <div className="col-meal">
                <h4 className="p-color">Price</h4>{meal.price}
              </div>
              <div className="col-meal">
                <br />
                <Link to={`/admin/edit/${meal.id}`}><em className="fa fa-edit p-color" />&nbsp;Edit</Link><br /> <br />
                <span onClick={() => this.onOpenModal(meal.id, meal.name, meal.image)} className="p-color delete" id={meal.name} role="button"><em className="fa fa-trash p-color" />&nbsp;Delete</span>
              </div>
            </div>
          </div>
        ))}
        <div className="meal-pagination margin-bottom">
          {this.props.meals.rows.length > 0 ?
            <Pagination
              activePage={this.state.activePage}
              itemsCountPerPage={limit}
              totalItemsCount={Math.ceil(this.props.meals.count)}
              pageRangeDisplayed={4}
              onChange={this.handlePageChange}
            />
          :
            <span> Meal List Empty</span>}
        </div>
      </div>
    );
  }
}
AllMeal.propTypes = {
  meals: PropTypes.object.isRequired,
  successMessage: PropTypes.object.isRequired,
  mealActions: PropTypes.object.isRequired
};

export const mapStateToProps = state => ({
  meals: state.meals,
  successMessage: state.successMessage
});

export const mapDispatchToProps = dispatch => ({
  mealActions: bindActionCreators(mealActions, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(AllMeal);
