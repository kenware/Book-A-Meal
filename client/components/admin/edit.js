import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Dropzone from 'react-dropzone';
import FormData from 'form-data';
import PropTypes from 'prop-types';

import * as mealActions from '../../redux/Action/mealAction';
import * as actions from '../../redux/Action/action';
import './index.scss';

export class Edit extends Component {
  constructor(props) {
    super(props);
    this.state = {
      meals: [],
      name: this.props.meals.name,
      price: this.props.meals.price,
      description: this.props.meals.description,
      file: [],
      updateMeal: 'Update Meal',
      validDescription: '',
      validPrice: '',
      validName: '',
      image: this.props.meals.image
    };
    this.onChange = this.onChange.bind(this);
    this.addMeal = this.addMeal.bind(this);
    this.onDrop = this.onDrop.bind(this);
  }

  /**
   * lifecycle hook called when component is mounted to DOM
   *
   * @memberof Admin index
   *
   * @returns {getAllMeal} fetches all meals
   */
  componentWillMount() {
    if (this.props.meals.name === '') {
      this.props.mealActions.getAllMeal();
    }
  }

  /**
   * lifecycle hook called when component receives props
   *
   * return new state after it is changed
   */
  static getDerivedStateFromProps(props, state) {
    if (props.meals !== state.meals) {
      return {
        meals: props.meals,
        image: props.meals.image,
        name: props.meals.name,
        price: props.meals.price,
        description: props.meals.description,
        updateMeal: 'Update Meal'
      };
    }
    return { meals: state.meals };
  }

  /**
   * remove error props in redux store if component unmount
   */
  componentWillUnmount() {
    this.props.actions.clearMessages();
  }
  /**
   * @param  {} e set state on input onchange event
  */

  onChange(e) {
    const { state } = this;
    state[e.target.name] = e.target.value;
    this.setState(state);
  }

  /**
   * @param  {} file sets image to state using drop box
  */
  onDrop(file) {
    this.setState({ file, image: '' });
  }

  /**
   * calls a redux function
   * adds a meal using form-data
  */
  addMeal(e) {
    e.preventDefault();
    const {
      name, price, description, file
    } = this.state;
    if (!name) { this.setState({ validName: 'Name is required' }); return; }
    if (!price) { this.setState({ validPrice: 'Price is required' }); return; }
    if (!description) { this.setState({ validDescription: 'Description is required' }); return; }
    const payload = new FormData();
    payload.append('name', name);
    payload.append('price', price);
    payload.append('description', description);
    file.forEach((photo) => {
      payload.append('file', photo);
    });
    // show loading icon on addmeal button
    this.setState({
      updateMeal: (<div><i className="fa fa-spinner fa-spin fa-2x fa-fw" aria-hidden="true" /></div>)
    });
    this.props.mealActions.updateMeal(this.props.meals.id, payload);
  }

  render() {
    const meal = this.props.meals;
    return (
      <div>
        <h2 style={{ marginTop: '2rem' }}>UPDATE A MEAL </h2>
        <div className="register-container" style={{ marginTop: '2px' }}>
          <div className="register-col" />
          <div className="register-wrapper" style={{ marginTop: '2px' }}>
            <h4 className="p-color text-center">Edit The Field Below To Update {meal.name} Meal</h4><br />
            <h3 className="text-center danger">{this.props.errorMessage.updateMealError }</h3>
            <h3 className="text-center danger">{this.state.message }</h3>
            <form>
              <div className="form-field">
                <label htmlFor="name">Name     <br />
                  <font color="red">{this.state.validName} </font>
                </label>
                <input onChange={this.onChange} type="text" value={this.state.name} name="name" id="name" required />
              </div>
              <div className="form-field">
                <label htmlFor="price">Price<br />
                  <font color="red">{this.state.validPrice} </font>
                </label>
                <input onChange={this.onChange} type="number" value={this.state.price} name="price" id="price" required />
              </div>
              <div className="form-field">
                <label htmlFor="name">Description<br />
                  <font color="red">{this.state.validDescription} </font>
                </label>
                <textarea cols="7" rows="9" onChange={this.onChange} className="form-label" name="description" value={this.state.description} id="description"required />

              </div>
              <div className="form-field">
                <label htmlFor="password">
                  Image
                </label>
                <span className="form-label">
                  <Dropzone
                    onDrop={this.onDrop}
                    accept="image/jpeg,image/jpg,image/tiff,image/gif,image/png,image/svg"
                    ref="dropzone"
                    multiple={false}
                  >
                    Drag and drop or click to select an image to upload.
                  </Dropzone>
                  {this.state.file.map(fil => <img key={fil.preview} src={fil.preview} className="img-fluid" alt="meal" />)}
                </span>
              </div>
              <div className="form-field">
                <button type="submit" className="button lg submit" onClick={this.addMeal}>{this.state.updateMeal}</button>
              </div>
              <div className="form-field">
                <span className="text-center form-label p-color">{this.props.successMessage.updateMealSuccess }</span>
              </div>
            </form>

          </div>
          <div className="register-col" />
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state, ownProps) => {
  let meals;
  if (state.meals.rows.length > 0) {
    meals = state.meals.rows.find(meal => meal.id === parseInt(ownProps.match.params.mealId));
  } else {
    meals = {
      name: '', price: 0, description: '', id: 0
    };
  }
  return {
    errorMessage: state.errorMessage,
    successMessage: state.successMessage,
    meals
  };
};

Edit.propTypes = {
  meals: PropTypes.object.isRequired,
  errorMessage: PropTypes.object.isRequired,
  successMessage: PropTypes.object.isRequired,
  mealActions: PropTypes.object.isRequired,
  actions: PropTypes.object.isRequired,
};

export const mapDispatchToProps = dispatch => ({
  mealActions: bindActionCreators(mealActions, dispatch),
  actions: bindActionCreators(actions, dispatch)
});
export default connect(mapStateToProps, mapDispatchToProps)(Edit);
