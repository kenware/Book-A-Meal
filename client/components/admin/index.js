import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Route } from 'react-router-dom';
import PropTypes from 'prop-types';
import Header from '../header/index';
import Header2 from '../header/header2';
import Footer from '../footer/index';
import * as actions from '../../redux/Action/action';
import * as mealActions from '../../redux/Action/mealAction';
import * as orderActions from '../../redux/Action/orderAction';
import './index.scss';
import Order from './order';
import Allmeals from './allMeals';
import Setmenu from './setMenu';
import Addmeals from './addMeal';
import EditMeal from './edit';
import history from '../../history';
import auth from '../../authenticate/auth';
import SideBar from './sidebar';

const limit = 6;
export class Admin extends Component {
  constructor(props) {
    super(props);
    this.state = {
      nav1: '',
      nav2: 'nav2',
      main: 'main-sidebar1',
      dash: 'dash',
      cPassword: 'cPassword',
      profile: 'profile',
      notific: 'notific',
      logout: 'logout',
      order: 'order',
      setmeal: 'setmeal',
      activePage: 1
    };
    this.logOut = this.logOut.bind(this);
    this.handleClose = this.handleClose.bind(this);
    this.handleClick = this.handleClick.bind(this);
    this.toggle = this.toggle.bind(this);
    this.handlePageChange = this.handlePageChange.bind(this);
  }
  /**
   * lifecycle hook called when component is mounted to DOM
   * @returns {mostOrder} load most ordered meal on page load
   * @returns {meals} get all meal
   * @returns {allOrder} get all orders.
   * refreshes token if it is not expired or logout and redirect to home if expired
   */
  componentDidMount() {
    this.props.mealActions.loadMostOrderedMeal(5);
    this.props.orderActions.getAllOrders();
    this.props.actions.refreshToken('admin');
  }
  // logout user
  logOut() {
    auth.logOut();
    history.push('/meals');
  }
  // toggle sidbar
  toggle() {
    if (this.state.nav1 === '') {
      this.setState({
        nav2: '',
        nav1: 'nav1',
        main: 'main-sidebar2'
      });
    } else {
      this.setState({
        nav2: 'nav2',
        nav1: '',
        main: 'main-sidebar1'
      });
    }
  }
  // Remove popover on mouseLeave
  handleClose(value) {
    const state = this;
    state[value] = value;
    this.setState(state);
  }
  // Popover on mouseEnter or hover
  handleClick(value) {
    const state = this;
    state[value] = '';
    this.setState(state);
  }

  handlePageChange(pageNumber) {
    const offset = (pageNumber - 1) * limit;
    this.setState({ activePage: pageNumber });
    this.props.orderActions.getAllOrders(limit, offset);
  }
  render() {
    const {
      nav1,
      nav2,
      main,
      dash,
      cPassword,
      profile,
      notific,
      logout,
      order,
      setmeal,
    } = this.state;
    const prop = {
      nav1,
      nav2,
      main,
      dash,
      cPassword,
      profile,
      notific,
      logout,
      order,
      setmeal,
    };

    return (
      <div>
        <span className="largeScreen-header">
          <Header component="admin" />
        </span>
        <span className="smallScreen-header">
          <Header2 />
        </span>
        <div className="admin-container">
          <SideBar
            {...prop}
            handleClick={this.handleClick}
            handleClose={this.handleClose}
            toggle={this.toggle}
            logOut={this.logOut}
          />
          <main className={`main ${this.state.main}`} id="main">
            <header className="header smallScreen-header">
              <div className="l-r-pad-text">
                <h4 className="p-color">ADMIN</h4>
              </div>
              <div>
                <h4 className="p-color">{window.localStorage.getItem('username')}</h4>
                <img src={window.localStorage.getItem('image')} className="user-img rounded-circle" alt="profile" />
              </div>
            </header>
            <Route
              exact
              path="/admin"
              render={props => (<Order
                mostOrder={this.props.mostOrder}
                allOrder={this.props.allOrder}
                activePage={this.state.activePage}
                handlePageChange={this.handlePageChange}
                {...props}
              />)}
            />
            <Route exact path="/admin/allmeals" component={Allmeals} />
            <Route exact path="/admin/setmenu" component={Setmenu} />
            <Route exact path="/admin/addmeals" component={Addmeals} />
            <Route path="/admin/edit/:mealId" component={EditMeal} />
            <Footer />
          </main>

        </div>

      </div>

    );
  }
}

Admin.propTypes = {
  mostOrder: PropTypes.array.isRequired,
  allOrder: PropTypes.object.isRequired,
  actions: PropTypes.object.isRequired,
  mealActions: PropTypes.object.isRequired,
  orderActions: PropTypes.object.isRequired,
};

export const mapStateToProps = state => ({
  errorMessage: state.errorMessage,
  mostOrder: state.mostOrder,
  allOrder: state.allOrder,
  cart: state.cart
});

export const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(actions, dispatch),
  mealActions: bindActionCreators(mealActions, dispatch),
  orderActions: bindActionCreators(orderActions, dispatch)
});
export default connect(mapStateToProps, mapDispatchToProps)(Admin);
