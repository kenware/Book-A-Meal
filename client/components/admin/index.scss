.form-label {
    flex:2;
    padding: 10px 10px 10px 10px;
    margin-right: 10px;
}
.detail-contents{
    display:flex;
    justify-content: space-around;
  }
  .detail-img{
    width:60%;
    height: 60%;
  }
  .detail-text{
   margin-top:1rem;
  }
  @media screen and (max-width: 600px) {
  .largeScreen-header{
     display: none;
    }
  }
  @media screen and (min-width: 600px) {
    .smallScreen-header{
       display: none;
      }
    }
    // react paginate components
    #react-paginate ul {
      display: inline-block;
      padding-left: 15px;
      padding-right: 15px;
  }
  
  #react-paginate li {
      display: inline-block;
  }
  
  #react-paginate .break a {
      cursor: default;
  }
