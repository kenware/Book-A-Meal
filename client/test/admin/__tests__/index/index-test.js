import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import { Admin, mapStateToProps, mapDispatchToProps } from '../../../../components/admin/index';
import { props, emptyProps } from '../../adminMockData';

describe('Test index Component of admin pages', () => {
  it('renders correctly', () => {
    const tree = shallow(<Admin {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should have 4 nested Route components', () => {
    const tree = mount(<MemoryRouter initialEntries={['/admin']}><Admin {...props} /></MemoryRouter>);
    expect(tree.find('Route')).toHaveLength(5);
  });

  it('should have respond to logOut method', () => {
    global.history.pushState = jest.fn();
    const handlePageChange = jest.spyOn(Admin.prototype, 'handlePageChange');
    const tree = shallow(<Admin {...props} />);
    const wrapper = tree.instance();
    wrapper.handlePageChange();
    expect(tree).toMatchSnapshot();
    expect(handlePageChange).toHaveBeenCalled();
  });

  it('should respond to mapStateToProps methods', () => {
    const ownProps = { match: { params: { mealId: 1 } } };
    const tree = mapStateToProps(props, ownProps);
    expect(tree).toMatchSnapshot();
  });

  it('should respond to mapDispatchToProps methods', () => {
    const tree = mapDispatchToProps(emptyProps);
    expect(tree).toMatchSnapshot();
  });
});
