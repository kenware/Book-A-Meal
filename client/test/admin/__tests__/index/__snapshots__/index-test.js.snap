// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test index Component of admin pages renders correctly 1`] = `
<div>
  <span
    className="largeScreen-header"
  >
    <Nav
      cart={Array []}
      component="admin"
      onModal={[Function]}
    />
  </span>
  <span
    className="smallScreen-header"
  >
    <Header />
  </span>
  <div
    className="admin-container"
  >
    <Sidebar
      cPassword="cPassword"
      dash="dash"
      handleClick={[Function]}
      handleClose={[Function]}
      logOut={[Function]}
      logout="logout"
      main="main-sidebar1"
      nav1=""
      nav2="nav2"
      notific="notific"
      order="order"
      profile="profile"
      setmeal="setmeal"
      toggle={[Function]}
    />
    <main
      className="main main-sidebar1"
      id="main"
    >
      <header
        className="header smallScreen-header"
      >
        <div
          className="l-r-pad-text"
        >
          <h4
            className="p-color"
          >
            ADMIN
          </h4>
        </div>
        <div>
          <h4
            className="p-color"
          />
          <img
            alt="profile"
            className="user-img rounded-circle"
            src={null}
          />
        </div>
      </header>
      <Route
        exact={true}
        path="/admin"
        render={[Function]}
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin/allmeals"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin/setmenu"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin/addmeals"
      />
      <Route
        component={[Function]}
        path="/admin/edit/:mealId"
      />
      <footer />
    </main>
  </div>
</div>
`;

exports[`Test index Component of admin pages should have respond handle click and close method 1`] = `
<div>
  <span
    className="largeScreen-header"
  >
    <Nav
      cart={Array []}
      component="admin"
      onModal={[Function]}
    />
  </span>
  <span
    className="smallScreen-header"
  >
    <Header />
  </span>
  <div
    className="admin-container"
  >
    <Sidebar
      cPassword="cPassword"
      dash="dash"
      handleClick={[Function]}
      handleClose={[Function]}
      logOut={[Function]}
      logout="logout"
      main="main-sidebar2"
      nav1="nav1"
      nav2=""
      notific="notific"
      order="order"
      profile="profile"
      setmeal="setmeal"
      toggle={[Function]}
    />
    <main
      className="main main-sidebar2"
      id="main"
    >
      <header
        className="header smallScreen-header"
      >
        <div
          className="l-r-pad-text"
        >
          <h4
            className="p-color"
          >
            ADMIN
          </h4>
        </div>
        <div>
          <h4
            className="p-color"
          />
          <img
            alt="profile"
            className="user-img rounded-circle"
            src={null}
          />
        </div>
      </header>
      <Route
        exact={true}
        path="/admin"
        render={[Function]}
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin/allmeals"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin/setmenu"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin/addmeals"
      />
      <Route
        component={[Function]}
        path="/admin/edit/:mealId"
      />
      <footer />
    </main>
  </div>
</div>
`;

exports[`Test index Component of admin pages should have respond handle click and close method 2`] = `
<div>
  <span
    className="largeScreen-header"
  >
    <Nav
      cart={Array []}
      component="admin"
      onModal={[Function]}
    />
  </span>
  <span
    className="smallScreen-header"
  >
    <Header />
  </span>
  <div
    className="admin-container"
  >
    <Sidebar
      cPassword="cPassword"
      dash="dash"
      handleClick={[Function]}
      handleClose={[Function]}
      logOut={[Function]}
      logout="logout"
      main="main-sidebar1"
      nav1=""
      nav2="nav2"
      notific="notific"
      order="order"
      profile="profile"
      setmeal="setmeal"
      toggle={[Function]}
    />
    <main
      className="main main-sidebar1"
      id="main"
    >
      <header
        className="header smallScreen-header"
      >
        <div
          className="l-r-pad-text"
        >
          <h4
            className="p-color"
          >
            ADMIN
          </h4>
        </div>
        <div>
          <h4
            className="p-color"
          />
          <img
            alt="profile"
            className="user-img rounded-circle"
            src={null}
          />
        </div>
      </header>
      <Route
        exact={true}
        path="/admin"
        render={[Function]}
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin/allmeals"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin/setmenu"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin/addmeals"
      />
      <Route
        component={[Function]}
        path="/admin/edit/:mealId"
      />
      <footer />
    </main>
  </div>
</div>
`;

exports[`Test index Component of admin pages should have respond handle click and close method 3`] = `
<div>
  <span
    className="largeScreen-header"
  >
    <Nav
      cart={Array []}
      component="admin"
      onModal={[Function]}
    />
  </span>
  <span
    className="smallScreen-header"
  >
    <Header />
  </span>
  <div
    className="admin-container"
  >
    <Sidebar
      cPassword="cPassword"
      dash="dash"
      handleClick={[Function]}
      handleClose={[Function]}
      logOut={[Function]}
      logout="logout"
      main="main-sidebar1"
      nav1="nav1"
      nav2="nav2"
      notific="notific"
      order="order"
      profile="profile"
      setmeal="setmeal"
      toggle={[Function]}
    />
    <main
      className="main main-sidebar1"
      id="main"
    >
      <header
        className="header smallScreen-header"
      >
        <div
          className="l-r-pad-text"
        >
          <h4
            className="p-color"
          >
            ADMIN
          </h4>
        </div>
        <div>
          <h4
            className="p-color"
          />
          <img
            alt="profile"
            className="user-img rounded-circle"
            src={null}
          />
        </div>
      </header>
      <Route
        exact={true}
        path="/admin"
        render={[Function]}
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin/allmeals"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin/setmenu"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin/addmeals"
      />
      <Route
        component={[Function]}
        path="/admin/edit/:mealId"
      />
      <footer />
    </main>
  </div>
</div>
`;

exports[`Test index Component of admin pages should have respond to logOut method 1`] = `
<div>
  <span
    className="largeScreen-header"
  >
    <Nav
      cart={Array []}
      component="admin"
      onModal={[Function]}
    />
  </span>
  <span
    className="smallScreen-header"
  >
    <Header />
  </span>
  <div
    className="admin-container"
  >
    <Sidebar
      cPassword="cPassword"
      dash="dash"
      handleClick={[Function]}
      handleClose={[Function]}
      logOut={[Function]}
      logout="logout"
      main="main-sidebar1"
      nav1=""
      nav2="nav2"
      notific="notific"
      order="order"
      profile="profile"
      setmeal="setmeal"
      toggle={[Function]}
    />
    <main
      className="main main-sidebar1"
      id="main"
    >
      <header
        className="header smallScreen-header"
      >
        <div
          className="l-r-pad-text"
        >
          <h4
            className="p-color"
          >
            ADMIN
          </h4>
        </div>
        <div>
          <h4
            className="p-color"
          />
          <img
            alt="profile"
            className="user-img rounded-circle"
            src={null}
          />
        </div>
      </header>
      <Route
        exact={true}
        path="/admin"
        render={[Function]}
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin/allmeals"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin/setmenu"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/admin/addmeals"
      />
      <Route
        component={[Function]}
        path="/admin/edit/:mealId"
      />
      <footer />
    </main>
  </div>
</div>
`;

exports[`Test index Component of admin pages should respond to mapDispatchToProps methods 1`] = `
Object {
  "actions": Object {
    "changePassword": [Function],
    "clearMessages": [Function],
    "getNotifications": [Function],
    "getUser": [Function],
    "loadErrorMessage": [Function],
    "loadNotifications": [Function],
    "loadSuccessMessage": [Function],
    "loadUser": [Function],
    "login": [Function],
    "refreshToken": [Function],
    "register": [Function],
    "resetLink": [Function],
    "updateProfile": [Function],
    "upgrade": [Function],
  },
  "mealActions": Object {
    "createMeal": [Function],
    "deleteMeal": [Function],
    "getAllMeals": [Function],
    "loadAllMeals": [Function],
    "loadErrorMessage": [Function],
    "loadMostOrdered": [Function],
    "loadMostOrderedMeal": [Function],
    "loadSuccessMessage": [Function],
    "updateMeal": [Function],
  },
  "orderActions": Object {
    "confirmStatus": [Function],
    "getAllOrders": [Function],
    "getMyOrder": [Function],
    "loadAllOrder": [Function],
    "loadErrorMessage": [Function],
    "loadMostOrdered": [Function],
    "loadMyOrder": [Function],
    "loadSuccessMessage": [Function],
    "loadTodayMenu": [Function],
    "orderMeal": [Function],
    "updateOrder": [Function],
  },
}
`;

exports[`Test index Component of admin pages should respond to mapStateToProps methods 1`] = `
Object {
  "allOrder": Object {
    "orders": Array [
      Object {
        "address": "Reverend Street",
        "createdAt": "2018-05-4",
        "id": 1,
        "meals": Array [
          Object {
            "descrption": "Very delicious",
            "id": 1,
            "name": "rice",
            "orderMealItems": Object {
              "quantity": 3,
              "totalPrice": 4,
            },
            "price": "200",
          },
        ],
        "quantity": 3,
        "status": "confirmed",
        "updatedAt": "2018-05-4",
        "user": Object {
          "id": 1,
          "image": "image",
          "name": "kevin",
        },
      },
    ],
  },
  "cart": undefined,
  "errorMessage": Object {
    "createMealError": "Wrong credentials",
  },
  "mostOrder": Array [
    Object {
      "description": "good",
      "id": 1,
      "image": "image",
      "name": "rice",
      "orderCount": 4,
      "price": "400",
    },
  ],
}
`;
